src/ima-state-machine.js

```diff
@@ -267,7 +267,7 @@ function onAdCompleted(options: Object, adEvent: any): void {
  */
 function onAdsCompleted(options: Object, adEvent: any): void {
   this.logger.debug(options.transition.toUpperCase());
-  if (this.playOnMainVideoTag() && this._contentComplete && this.config.reloadSourceOnComplete) {
+  if (this.playOnMainVideoTag() && this._contentComplete && !this.player.config.playback.playAdsWithMSE) {
```

Comment:
Don’t use nested player configs in the plugins, in general it will enable separating player and plugins logic if and when required.
What is the benefit of removing the plugin config?

Reply:
The benefit is to be sure that the reload will happened once this config will be false, without handling more config.

Reply:
@OrenMe you need this config in each ad plugin - ima, bumper and maybe dai